### PR TITLE
Fixed checking out non-existent branch

### LINKED ISSUE #549: Switching to a branch that doesn't exist from menu breaks management portal
If you type-in a branch that does not exist into the "Checkout existing branch" , the management portal stalls and eventually stops responding. Users should get some sort of feedback that the checkout failed because the branch does not exist

### PR DESCRIPTION
**Pull request by isc-etamarch**
Fixes #549, #539

@isc-tleavitt Right now this throws an error (so that user is aware that checkout failed). Is there a better way to show an alert on the management portal / vscode / studio? (This happens on "AfterUserAction" so we have no access to the "Target" functionality of %Studio.SourceControl.Base.

### CODE REVIEW

File: cls/SourceControl/Git/Utils.cls [hunk 1]
```diff
@@ -327,7 +327,10 @@ ClassMethod AfterUserAction(Type As %Integer, Name As %String, InternalName As %
         }
     } elseif (menuItemName = "SwitchBranch") {
         if (Answer = 1) {
-            do ..SwitchBranch(Msg)
+            set status = ..SwitchBranch(Msg)
+            if $$$ISERR(status) {
+                Quit status
```
Comment (isc-tleavitt) [inline cls/SourceControl/Git/Utils.cls:332]: Quitting with an error %Status is the right approach if something goes wrong in AfterUserAction; short of changing to a CSP page like sync I don't see a way to get a better UX than with this.

File: cls/SourceControl/Git/Utils.cls [hunk 2]
```diff
@@ -400,6 +403,13 @@ ClassMethod NewBranch(newBranchName As %String) As %Status
 
 ClassMethod SwitchBranch(targetBranchName As %String) As %Status
 {
+    // Make sure that branch exists first
```
Comment (isc-tleavitt) [inline cls/SourceControl/Git/Utils.cls:406]: What is current behavior (prior to this change) if you pick a branch that already exists on the remote but isn't local yet? What is the behavior after this change?

**Review by isc-tleavitt — CHANGES_REQUESTED** [review]

File: cls/SourceControl/Git/Utils.cls (hunk 2 above)
Comment (isc-etamarch) [reply cls/SourceControl/Git/Utils.cls]: Now both should succeed

**Comment by isc-etamarch** [comment]
I linked #539 since it was related and a quick addition to the current change. Now behavior is that checking out remote or local branches succeed unless they don't exist or you have uncommitted changes that will be deleted.

**Review by isc-tleavitt — APPROVED** [review]

**Pull request merged**